src/tairzset.c

```diff
@@ -1722,7 +1728,99 @@ int exZpopMinMaxGenericCommand(RedisModuleCtx *ctx, RedisModuleString **argv, in
         return REDISMODULE_OK;
     }
 
-    exGenericZpopCommand(ctx, key, where, count); 
+    exGenericZpopCommand(ctx, key, where, NULL, count); 
+
+    return REDISMODULE_OK;
+}
+
+typedef struct exBzpopInfo {
+    int where;
+} ExBzpopInfo;
+
+int exBzpopReplyCallback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    RedisModuleString *key_str = RedisModule_GetBlockedClientReadyKey(ctx);
+    RedisModuleKey *key = RedisModule_OpenKey(ctx, key_str, REDISMODULE_WRITE);
+    int type = RedisModule_KeyType(key);
+    if (REDISMODULE_KEYTYPE_EMPTY != type && RedisModule_ModuleTypeGetType(key) != TairZsetType) {
+        return REDISMODULE_ERR;
+    }
+
+    if (type == REDISMODULE_KEYTYPE_EMPTY || key == NULL) {
+        return REDISMODULE_ERR;
+    }
+    ExBzpopInfo *info = RedisModule_GetBlockedClientPrivateData(ctx);
+
+    exGenericZpopCommand(ctx, key, info->where, key_str, 1);
+    return REDISMODULE_OK;
+}
+
+void freePrivdata(RedisModuleCtx *ctx,void *privdata) {
+    RedisModule_Free(privdata);
+}
+
+int exBzpopTimeoutCallback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc)  {
+    return RedisModule_ReplyWithNull(ctx);
+}
+
+/* EXBZPOPMIN / EXBZPOPMAX actual implementation. */
+int exBzpopMinMaxGenericCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, int where) {
+    RedisModule_AutoMemory(ctx);
+
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    double ftval;
+    long long timeout;
+    RedisModuleKey *key;
+    TairZsetObj *o;
+    int j;
+
+    if (RedisModule_StringToDouble(argv[argc - 1], &ftval) != REDISMODULE_OK) {
+        RedisModule_ReplyWithError(ctx, "ERR timeout is not an float or out of range");
+        return REDISMODULE_OK;
+    }
+
+    timeout = (long long) (ftval * 1000.0);
+
+    if (timeout < 0) {
+        RedisModule_ReplyWithError(ctx, "ERR timeout is negative");
+        return REDISMODULE_OK;
+    }
+
+    for (j = 1; j < argc - 1; j++) {
+        key = RedisModule_OpenKey(ctx, argv[j], REDISMODULE_WRITE);
+        int type = RedisModule_KeyType(key);
+
+        if (type == REDISMODULE_KEYTYPE_EMPTY || key == NULL) continue;
+
+        if (REDISMODULE_KEYTYPE_EMPTY != type && RedisModule_ModuleTypeGetType(key) != TairZsetType) {
+            RedisModule_ReplyWithError(ctx, REDISMODULE_ERRORMSG_WRONGTYPE);
+            return REDISMODULE_OK;
+        }
+
+        o = RedisModule_ModuleTypeGetValue(key);
+        if (o == NULL) continue;
+        long llen = exZsetLength(o);
+        if (llen == 0) continue;
+
+        exGenericZpopCommand(ctx, key, where, argv[j], 1);
```

Comment:
When you write the `EXBZPOPMIN / EXBZPOPMAX` to AOF and Slave, u should rewrite them to an unblocked command (ZPOPXX etc.)，not just simple use `RedisModule_ReplicateVerbatim`.

Reply:
Oh. I misunderstood the functionality of `RedisModule_ReplicateVerbatim`.